### PR TITLE
fix(curriculum): added test case and resorted tests for arguments-optional page

### LINKED ISSUE #50402: Arguments Optional - Clearer test case ordering
### Describe the Issue

Our test cases for this should add a test to more clearly indicate what happens if only one argument is used and that argument is a valid number. Also, we should re-order the tests for clarity.

### Affected Page

https://www.freecodecamp.org/learn/javascript-algorithms-and-data-structures/intermediate-algorithm-scripting/arguments-optional

### Your code

Verbose solution from the guide that highlights the possible cases
```javascript
function addTogether() {
  const [first, second] = arguments;
  // First argument is not a number
  if (typeof(first) !== "number") {
    return undefined;
  }
  // First argument is a number
  //  and second argument is not defined
  else if (arguments.length === 1) {
    function addSecond(second) {
      // New argument is not a number
      if (typeof(second) !== "number") {
        return undefined;
      }
      // New argument is a number
      else {
        return first + second;
      }
    }
    // Note: returning a *function*
    return addSecond;
  }
  // First argument is a number
  //  and second argument is not a number
  else if (typeof(second) !== "number") {
    return undefined;
  }
  // First argument is a number
  //  and second argument is a number
  else {
    return first + second;
  }
}
```

### Expected behavior

Our test cases should more clearly map to the possible cases:

https://github.com/freeCodeCamp/freeCodeCamp/tree/main/curriculum/challenges/english/02-javascript-algorithms-and-data-structures/intermediate-algorithm-scripting

---

Updated tests proposed:

# --hints--

`addTogether(2, 3)` should return 5.

```js
assert.deepEqual(addTogether(2, 3), 5);
```

`addTogether(23, 30)` should return 53.

```js
assert.deepEqual(addTogether(23, 30), 53);
```

`addTogether("2", 3)` should return `undefined`.

```js
assert.isUndefined(addTogether('2', 3));
```

`addTogether(5, undefined)` should return `undefined`.

```js
assert.isUndefined(addTogether(5, undefined));
```

`addTogether("https://www.youtube.com/watch?v=dQw4w9WgXcQ")` should return `undefined`.

```js
assert.isUndefined(addTogether('https://www.youtube.com/watch?v=dQw4w9WgXcQ'));
```

`addTogether(5)` should return a function.

```js
assert.deepEqual(typeof(addTogether(5)), 'function');
```

`addTogether(5)(7)` should return 12.

```js
assert.deepEqual(addTogether(5)(7), 12);
```

`addTogether(2)([3])` should return `undefined`.

```js
assert.isUndefined(addTogether(2)([3]));
```

`addTogether(2, "3")` should return `undefined`.

```js
assert.isUndefined(addTogether(2, '3'));
```

### Screenshots

_No response_

### System

All

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by Farissoliman**
Checklist:

- [x] I have read and followed the [contribution guidelines](https://contribute.freecodecamp.org).
- [x] I have read and followed the [how to open a pull request guide](https://contribute.freecodecamp.org/#/how-to-open-a-pull-request).
- [x] My pull request targets the `main` branch of freeCodeCamp.
- [x] I have tested these changes either locally on my machine, or GitPod.

Closes #50402

### CODE REVIEW

**Comment by Farissoliman** [comment]
Checklist:



- [x] I have read and followed the [contribution guidelines](https://contribute.freecodecamp.org).
- [x] I have read and followed the [how to open a pull request guide](https://contribute.freecodecamp.org/#/how-to-open-a-pull-request).
- [x] My pull request targets the `main` branch of freeCodeCamp.
- [x] I have tested these changes either locally on my machine, or GitPod.



Closes #50402


Fixed a failing test case from previous commit. All tests now passing

File: curriculum/challenges/english/02-javascript-algorithms-and-data-structures/intermediate-algorithm-scripting/arguments-optional.md [hunk 1]
```diff
@@ -24,28 +24,30 @@ If either argument isn't a valid number, return undefined.
...
+`addTogether(5)` should return another function.
 
 ```js
-assert.deepEqual(addTogether(5)(7), 12);
+assert.deepEqual(typeof(addTogether(5)), 'function');
 ```
 
-`addTogether("https://www.youtube.com/watch?v=dQw4w9WgXcQ")` should return `undefined`.
+
+`addTogether(5)` should return another function then return sum if the other function has a parameter. should return 12.
```
Comment (jeremylt) [inline curriculum/challenges/english/02-javascript-algorithms-and-data-structures/intermediate-algorithm-scripting/arguments-optional.md:46]: Also, this isn't the wording requested.

Please use the ordering and text given in the issue.

**Review by gikf — APPROVED** [review]
LGTM :+1: 

---

Note for future - it is recommended to make changes on a separate branch, than the `main` branch. See more about it in point 3 at https://contribute.freecodecamp.org/#/how-to-contribute-to-the-codebase?id=contributing-to-the-codebase.

**Review by jeremylt — APPROVED** [review]
Lgtm

**Pull request merged**